### PR TITLE
feature: Add min batch size support

### PR DESCRIPTION
**Pull request by Ehud-Lev-Forter**
**Add min batch size support**
We would like to support minimum batch size. Since we might have a time with no traffic and we don't want to wait forever for the events that are not satisfy minimum batch, we will also add `minBatchTimeoutInMillis`, if we passed this timeout, code will be as today.
[see issue](https://github.com/confluentinc/parallel-consumer/issues/560)

### Checklist

- [ ] Documentation (if applicable)
- [ ] Changelog

### CODE REVIEW

**Comment by astubbs** [comment]
Title should be something likes:

feature: Add min batch size support

**Review by rkolesnev — CHANGES_REQUESTED** [review]
In general logical flow looks fine to me.
I would really like to see more tests and integration tests for it though - with transactions, different ordering modes (Unordered / Key /Partition) and 'normal' non-reactor usage mode of PC - it has a potential to introduce edge case issues that additional tests could shake out.

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/ParallelConsumerOptions.java [hunk 1]
```diff
@@ -393,6 +419,23 @@ public void validate() {
...
 
         transactionsValidation();
+        validateMinBatchParameters();
+    }
+
+    private void validateMinBatchParameters() {
+        if (isEnforceMinBatch()){
+            if (minBatchSize > batchSize)
+                throw new IllegalArgumentException(
+                        msg("minBatchSize can not by bigger than batchSize: {} > {}",
```
Comment (eddyv) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/ParallelConsumerOptions.java:429]:
```suggestion
                        msg("minBatchSize cannot be bigger than batchSize: {} > {}",
```
typo

**Review by eddyv — COMMENTED** [review]
Seems to break a test 

```
[ERROR] io.confluent.parallelconsumer.state.WorkManagerTest.resumesFromNextShard(ProcessingOrder)[1] -- Time elapsed: 0.007 s <<< FAILURE!
14:36:51  expected: 1
14:36:51  but was : 2
```
